### DISCUSSION TITLE
How to set basePath for application

### DISCUSSION THREAD

**Discussion by Schetnan** [nuxt/nuxt#17091] [Questions] [answered]
In Nuxt2 we have the ability to set the base path ( contextRoot ) of our application in the following manner:

  router: {
    base: '/my/app',
  }

I am wondering what the equivalent configuration is in nuxt3?

**Comment by wobsoriano** [comment, marked as answer]
Have you tried this?

```ts
import { defineNuxtConfig } from 'nuxt3'

export default defineNuxtConfig({
    router: {
        base: '/my/app'
    }
})
```

**Reply by Schetnan** [reply]
I did.  It doesn't seem to work.  The router element has been removed from the documentation of nuxt.config.ts - so it seems that maybe the router configuration is no longer valid in nuxt3?

**Reply by Schetnan** [reply]
@wobsoriano  It turns out that setting the router base does work as long as there is a an index.vue file in the pages directory.  It doesn't work with only an app.vue.   Marked as answer.  Thank you!

**Reply by lucassimines** [reply]
> @wobsoriano It turns out that setting the router base does work as long as there is a an index.vue file in the pages directory. It doesn't work with only an app.vue. Marked as answer. Thank you!

I can't make it work with `/admin` base, the `assets` don't load correctly

**Reply by syifeng** [reply]
Got the same question. 
nuxt-version: v3.0.0-27306529.1ef6e3f
code with 
```
import { defineNuxtConfig } from 'nuxt3'

export default defineNuxtConfig({
    router: {
        base: '/my/app'
    }
})
```
not worked

**Reply by lkononenko** [reply]
@syifeng there is possibility to set baseUrl - https://v3.nuxtjs.org/docs/directory-structure/nuxt.config#baseurl

**Comment by danielroe** [comment]
As @lkononenko [says](https://github.com/nuxt/framework/discussions/1525#discussioncomment-2361673), the solution (now) is to set `app.baseURL`:

```js
import { defineNuxtConfig } from 'nuxt3'

export default defineNuxtConfig({
  app: {
    baseURL: '/my/app'
  }
})
```

**Reply by justindasilva** [reply]
@danielroe 
I find that there is some strange behaviour happening when I use that option.

My repo has following structure:

```
<root>
- /app
- /docs
```

A Nuxt 2 + Bridge projects lives under `/app`
A Nuxt 3 + Nuxt Content project lives under `/docs`

My end goal is that every time my `app` is deployed somewhere, part of my build process is to run `yarn generate` in the `/docs` folder and copy the contents of `.output/public` over to `/app/public/docs`

The idea is that I want my `docs` to be accessible after any deployment at `/docs`.

I've tried setting this option in `nuxt.config.ts` in `/docs`
```
app: {
  baseURL: '/docs'
},
```

I'm getting very strange behavior where it generates a folder called `docs` within the generated folder. But there are files that live outside that folder that the `docs` needs (assets, etc.)

Given the app structure I have defined, is there a better, more ideal way to accomplish what I'm after here? I kinda just assumed that doing a `yarn generate` on my docs and throwing that static folder into the `public` folder of my app before doing a `yarn build` would yield the best result.

**Reply by danielroe** [reply]
Yes, that should work. (Bear in mind you won't be able to use NuxtLinks to `/docs` URLs - or you'll need to add a router hook to force them to fully reload the page.)

There is currently one snag - and you can follow that issue at https://github.com/unjs/nitro/pull/329.

**Reply by justindasilva** [reply]
@danielroe 
I took the time to create a reproduction repo. I tried to explain things as best as I could in the README file. Do you mind taking a look:

https://github.com/justindasilva/nuxt-plus-docs

As mentioned, definitely something strange going on when using `yarn build` (works fine with `yarn dev`)

**Reply by mybearworld** [reply]
I am a *little* late here, but that doesn't work for me either, everything still assumes the site is hosted on `/` with:
```ts
// https://nuxt.com/docs/api/configuration/nuxt-config
export default defineNuxtConfig({
  app: {
    baseURL: "/lorem/",
  },
});
```
I have also tried app.cdnURL and vite.base, none of which worked.

**Reply by mpgalaxy** [reply]
Hi, even a little later, BUT: This still is an Issue in production.
Using
``` 
app: {
    baseURL: "/test/",
  },
``` 
in nuxt.config this works in dev environment, BUT it fails in production even with 
` "build": "NUXT_APP_BASE_URL=/test/ nuxt build" ` and
` "start": "NUXT_APP_BASE_URL=/test/ node .output/server/index.mjs" `

I get 404s on images and relative paths like /dashboard, somehow this still defaults to "/" in production.

Maybe related to https://github.com/nuxt/nuxt/issues/28474 without any working solution
